server/src/main/java/org/opensearch/action/bulk/BulkRequestParser.java

```diff
@@ -78,6 +78,33 @@ public final class BulkRequestParser {
     private static final ParseField IF_PRIMARY_TERM = new ParseField("if_primary_term");
     private static final ParseField REQUIRE_ALIAS = new ParseField(DocWriteRequest.REQUIRE_ALIAS);
 
+    private enum Action {
```

Comment:
Seems like we could make it as simple as:

```
    private enum Action {
        CREATE,
        DELETE,
        INDEX,
        UPDATE;

        static Action parse(String name, int line) {
            return Arrays
                .stream(values())
                .filter(a -> a.name().equalsIgnoreCase(name))
                .findAny()
                .orElseThrow(() -> new IllegalArgumentException(
                        "Unknown action line [" + line + "], expected one of [create, delete, index, update]" + " but found [" + name + "]"));
        }
    }
```

Reply:
(or replace `Arrays,stream(values())` with `for(Action a: values()) { ... }`

Reply:
Thanks @adnapibar, sounds like micro-optimization but good to know), do we really need an enum there? it looks like 4 string constants would be more than  sufficient ...

Reply:
Not really, we can use string constants if you prefer so?

Reply:
I think it would be much simpler since all changes are contained in the single class